credentialz/credentialz.proto

```diff
@@ -582,14 +585,36 @@ message RotateHostCredentialsResponse {
   }
 }
 
+// KeySuite specifies the valid types of keys and their lengths that
+// can be used.
+enum KeyGen {
+  KEY_GEN_SSH_KEY_UNSPECIFIED = 0;
+  KEY_GEN_SSH_KEY_TYPE_RSA_2048 = 1;
+  KEY_GEN_SSH_KEY_TYPE_ECDSA_P_256 = 2;
+  KEY_GEN_SSH_KEY_TYPE_ECDSA_P_521 = 3;
+  KEY_GEN_SSH_KEY_TYPE_EDDSA_ED25519 = 4;
+}
+
+// A request to ask the target if it can generate key pairs
+message CanGenerateKeyRequest {
```

Comment:
Can you say why this request/response is necessary?

From the operator side I imagined we would say:
  "Here are your public keys: <set of keys>"
  
 I hadn't seen the need to ask:
  "Can you make an X?"

Reply:
Perhaps this is ssh-keys and cert-keys being crossed up by bad naming choices :(

(though I still imagine you would just dictate the key/cert types to the device)

Reply:
We wanted to provide a way for the operator to generate the keys on the box instead of externally generating them, but also give an option to inject externally generated keys. So based on the discussion in https://github.com/openconfig/gnsi/issues/58, we have added some RPC, messages to support this. This API is used to check the key generation capability of the switch.